oms-azure-storage-analytics-solution/scripts/AzureSAIngestionLogs-MS-Mgmt-SA.ps1

```diff
@@ -376,110 +376,114 @@ function Cleanup-Variables {
 #Authenticate to Azure with SPN section
 "Logging in to Azure..."
 $ArmConn = Get-AutomationConnection -Name AzureRunAsConnection 
-$AsmConn = Get-AutomationConnection -Name AzureClassicRunAsConnection  
 
-if ([string]::IsNullOrEmpty($SubscriptionidFilter)) {
-    $subscriptionid = $ArmConn.SubscriptionId
+if ($ArmConn  -eq $null)
+{
+	throw "Could not retrieve connection asset AzureRunAsConnection,  Ensure that runas account  exists in the Automation account."
 }
-Else {$subscriptionid = $SubscriptionidFilter}
-
 
 # retry
 $retry = 6
 $syncOk = $false
-do { 
-    try {  
-        Add-AzureRMAccount -ServicePrincipal -Tenant $ArmConn.TenantID -ApplicationId $ArmConn.ApplicationID -CertificateThumbprint $ArmConn.CertificateThumbprint
-        $syncOk = $true
-    }
-    catch {
-        $ErrorMessage = $_.Exception.Message
-        $StackTrace = $_.Exception.StackTrace
-        Write-Warning "Error during sync: $ErrorMessage, stack: $StackTrace. Retry attempts left: $retry"
-        $retry = $retry - 1       
-        Start-Sleep -s 60        
-    }
+do
+{ 
+	try
+	{  
+		Add-AzureRMAccount -ServicePrincipal -Tenant $ArmConn.TenantID -ApplicationId $ArmConn.ApplicationID -CertificateThumbprint $ArmConn.CertificateThumbprint
```

Comment:
This is also Environment specific can we add the env param?

Reply:
Sorry I dont get it , What shall go into env variable here ?
Add-AzureRMAccount -ServicePrincipal -Tenant $ArmConn.TenantID -ApplicationId $ArmConn.ApplicationID -CertificateThumbprint $ArmConn.CertificateThumbprint

Reply:
We're hardcoding endpoints (in AquireToken) and logging in to the AzureCloud environment unconditionally - this will fail when used in other clouds (Gov, Germany, China) - we should parameterize these values